src/parser/mod.rs

```diff
@@ -5589,7 +5589,23 @@ impl<'a> Parser<'a> {
         let clustered_by = self.parse_optional_clustered_by()?;
         let hive_formats = self.parse_hive_formats()?;
         // PostgreSQL supports `WITH ( options )`, before `AS`
-        let with_options = self.parse_options(Keyword::WITH)?;
+        let mut with_options: Vec<SqlOption> = vec![];
+        let mut order_exprs: Vec<Vec<OrderByExpr>> = vec![];
+        if self.parse_keyword(Keyword::WITH) {
+            if self.parse_keyword(Keyword::ORDER) {
```

Comment:
```suggestion
        if self.parse_keywords(&[Keyword::WITH, Keyword::ORDER]) {
```
Also I think we can keep the postgres path separate since this feature doesn't apply to that dialect? - so that this part remains the same
```rust
        let with_options = self.parse_options(Keyword::WITH)?;
```
We can make the fuctionality conditional with e.g. `if self.dialect.supports_with_order()` or similar method